src/components/dialog/dialog.tsx

```diff
@@ -100,6 +100,7 @@ export class Dialog {
 
     public componentWillLoad() {
         this.id = createRandomString();
+        this.showFooter = !!this.host.querySelector('[slot="button"]');
```

Comment:
@adrianschmidt Of course! It's when you use the component within a "dialog component" with the dialog service in Lime CRM and you don't have a toggle for show/hide. Here's the code from the usage where it breaks in the CRM solution:

```
public render() {
        return (
            <limel-dialog
                open={true}
                onClose={this.closeDialog}
                fullscreen={true}
            >
                <lwc-limepkg-workorder-resource-planner
                    platform={this.platform}
                    context={this.context}
                ></lwc-limepkg-workorder-resource-planner>
            </limel-dialog>
        );
    }
```

The reason you don't see it in the dialog examples here is that we have no example where `open` is `true` from the start.

If you change `render()` in `examples/dialog.tsx` to the following you will see the footer being rendered even without providing a slot for the button.

[examples/dialog.tsx render()](https://github.com/Lundalogik/lime-elements/blob/07ce600a0f3300fa5c4587d5e07be8e7a18f0619/src/components/dialog/examples/dialog.tsx#L16-L30)
```
    public render() {
        return [
            <limel-dialog open={true} onClose={this.closeDialog}>
                <p>This is a simple alert-dialog.</p>
            </limel-dialog>,
        ];
    }
```

It's best to see this locally in your dev tools but if you set some breakpoints where `showFooter` is set and in `renderFooter` you will notice that the breakpoint in `renderFooter` hits first and will render the `<footer>` (because `this.showFooter = true` in the constructor). Then the breakpoint in `initialize()` is hit, being called from `componentDidLoad()` (called after `render()`).

With the code in this PR your breakpoint on `if(this.showFooter) {` in `renderFooter()` will have `this.showFooter` = `false`, while it will be `true` in `main`.

Here's a screen recording of this behavior showing the footer in the modified example provided above.
Note that it's not the best example with the breakpoints since the other examples has showFooter set to `true` and was hit when I recorded it.

https://user-images.githubusercontent.com/435885/117134126-74758500-ada5-11eb-82dd-8e273e01b371.mov